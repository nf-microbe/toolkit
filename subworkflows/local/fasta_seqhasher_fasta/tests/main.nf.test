nextflow_workflow {

    name "Test Subworkflow FASTA_SEQHASHER_FASTA"
    script "../main.nf"
    workflow "FASTA_SEQHASHER_FASTA"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fasta_seqhasher_fasta"
    tag "csvtk"
    tag "csvtk/concat"
    tag "seqhasher"
    tag "seqhasher/seqhasher"
    tag "seqhasher/sequniq"
    tag "seqhasher/sequniq"
    tag "seqkit"
    tag "seqkit/grep"

    test("b_fragilis fasta.gz") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test1', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                    ],
                    [
                        [ id:'test2', single_end:false ], // meta map
                        file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test2.contigs.fa.gz", checkIfExists: true)
                    ]
                )
                input[1] = file("${projectDir}/bin/seq-hasher", checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
