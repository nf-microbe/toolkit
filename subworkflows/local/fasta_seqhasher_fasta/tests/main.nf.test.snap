{
    "b_fragilis fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1.fa.gz:md5,3d5d63cdeeda1ea7299a68d674bf817c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3ada7082bf6dff36f6c62264c7264ca4",
                    "versions.yml:md5,556aaad95a18aa6f9675af833b7c94ff",
                    "versions.yml:md5,7c10d5ad7107a75066ec30828e696ff1",
                    "versions.yml:md5,a9b1723acacf301e38c54cd33ab37363"
                ],
                "unique_seqs_fasta_gz": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1.fa.gz:md5,3d5d63cdeeda1ea7299a68d674bf817c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3ada7082bf6dff36f6c62264c7264ca4",
                    "versions.yml:md5,556aaad95a18aa6f9675af833b7c94ff",
                    "versions.yml:md5,7c10d5ad7107a75066ec30828e696ff1",
                    "versions.yml:md5,a9b1723acacf301e38c54cd33ab37363"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-02T06:50:40.334413044"
    }
}